src/MyEventEmitter.js

```diff
@@ -1,14 +1,102 @@
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.listeners = {};
+  }
+
+  emptyListener(eventName) {
+    if (!this.listeners[eventName]) {
+      this.listeners[eventName] = [];
+    }
+  }
+
+  setValidation(eventName, eventListener) {
+    if (
+      !(typeof eventName === 'string' && typeof eventListener === 'function')
+    ) {
+      throw new Error('ERROR: invalid arguments');
+    }
+  }
+
+  onceArgs(eventName, eventListener) {
+    const event = (...args) => {
+      eventListener(...args);
+      this.off(eventName, event);
+    };
+
+    return event;
+  }
+
+  on(eventName, eventListener) {
+    this.setValidation(eventName, eventListener);
+    this.emptyListener(eventName);
+
+    this.listeners[eventName].push(eventListener);
+
+    return this;
+  }
+
+  once(eventName, eventListener) {
+    this.setValidation(eventName, eventListener);
+    this.emptyListener(eventName);
+
+    this.listeners[eventName].push(this.onceArgs(eventName, eventListener));
```

Comment:
add check if such listener exists

Reply:
MEIN GOTT, nobody had so many comments in that task such as me😂